pkg/push/api/api.go

```diff
@@ -561,7 +569,136 @@ func (s *PushService) UpdatePush(
 		}
 		return nil, dt.Err()
 	}
-	return &pushproto.UpdatePushResponse{}, nil
+
+	if updatedPushPb != nil {
+		// For security reasons we remove the service account from the API response
+		updatedPushPb.FcmServiceAccount = ""
+	}
+	return &pushproto.UpdatePushResponse{
+		Push: updatedPushPb,
+	}, nil
+}
+
+func (s *PushService) updatePushNoCommand(
+	ctx context.Context,
+	req *pushproto.UpdatePushRequest,
+	localizer locale.Localizer,
+	editor *eventproto.Editor,
+) (*pushproto.UpdatePushResponse, error) {
+	if err := s.validateUpdatePushRequestNoCommand(ctx, req, localizer); err != nil {
+		return nil, err
+	}
+	var updatedPushPb *pushproto.Push
+	var updatePushTagsEvent *eventproto.Event
+	var updatePushNameEvent *eventproto.Event
+	tx, err := s.mysqlClient.BeginTx(ctx)
+	if err != nil {
+		s.logger.Error(
+			"Failed to begin transaction",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.Error(err),
+			)...,
+		)
+		dt, err := statusInternal.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalize(locale.InternalServerError),
+		})
+		if err != nil {
+			return nil, statusInternal.Err()
+		}
+		return nil, dt.Err()
+	}
+	err = s.mysqlClient.RunInTransaction(ctx, tx, func() error {
+		pushStorage := v2ps.NewPushStorage(tx)
+		push, err := pushStorage.GetPush(ctx, req.Id, req.EnvironmentNamespace)
+		if err != nil {
+			return err
+		}
+		prev := &domain.Push{}
+		if err = copier.Copy(prev, push); err != nil {
+			return err
+		}
+		push.Name = req.Name
+		push.Tags = req.Tags
+		updatedPushPb = push.Push
+
+		updatePushNameEvent, err = domainevent.NewEvent(
```

Comment:
We can't have more than one event being reported per request.

Reference:
https://github.com/bucketeer-io/bucketeer/blob/6dccbca812e60f9ae5f1250906a24f300646c209/pkg/feature/api/feature.go#L792-L806